### DISCUSSION TITLE
Network requirements

### DISCUSSION THREAD

**Discussion by dcbrown16** [Azure/azure-dev#3161] [General]
Is it possible to install and use azd commands in an environment with no network connectivity?
I assume that GitHub connectivity is required for the installation, and some Entra ID endpoints would be required to log in. Are the main commands then connecting to management.azure.com, like the Azure REST API, or is there some other endpoint? Any list of required URLs?
If we can download the files somewhere and then copy them to the environment with no Internet, info about what files to copy would be useful.

**Comment by vhvb1989** [comment]
What `azd commands` are you trying to run w/o network?

For the main commands (provision, deploy), azd needs to connect to Azure.

**Reply by dcbrown16** [reply]
Thanks for checking Victor, I'm actually in support and researching this question for a customer. I believe they have a closed network environment for security purposes, so they are trying to assess whether they will be able to run azd commands there.
They mentioned that they had set up an environment locally using azd, I believe using OpenAI on their data. They wanted to do the same thing on Azure VMs with no internet access.
Are there any commands that they could run if they copy over some kind of packages to the VMs?

**Reply by vhvb1989** [reply]
> on Azure VMs with no internet access.

azd requires connection to a few endpoints.  It won't work properly without that. I don't think you can have a VM without network, as there's has to be a way how you connect to that VM, right? 

You might want to take a look to this azure-sample: https://github.com/Azure/GPT-RAG
It has a zero-trust model you can turn on which places all your resources under private endpoints and creates a VM which is only accessible from Bastion.
On this model, the VM has access to internet and to the private endpoints, so the deployment is only accessible from that VM.

**Reply by dcbrown16** [reply]
Sorry for the unclear point, the customer wants to restrict the outbound connectivity. Presumably they would have a way to connect to the VM through ExpressRoute, VPN etc. Are you able to provide any info about the required outbound connectivity for azd commands? Is it mainly connecting to management.azure.com, or some other endpoint(s)?
